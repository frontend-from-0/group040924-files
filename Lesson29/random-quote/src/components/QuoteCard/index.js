import './styles.css';
// export const QuoteCard = (props) => {

//   return (
//     <section>
//       <p>{props.quote}</p>
//       <p>{props.author}</p>
//     </section>
//   );
// };

export const QuoteCard = ({quote, author}) => {

  return (
    <section className="QuoteCard">
      <p>{quote}</p>
      <p>{author}</p>
    </section>
  );
};