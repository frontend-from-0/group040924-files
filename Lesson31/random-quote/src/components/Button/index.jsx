
export const Button = ({ label, handleOnClick }) => {
  return <button className="btn" onClick={handleOnClick}>{label}</button>;
};
