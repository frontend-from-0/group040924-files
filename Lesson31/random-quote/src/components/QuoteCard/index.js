import './styles.css';

export const QuoteCard = ({quote, author}) => {

  return (
    <section className="QuoteCard">
      <p>{quote}</p>
      <p>{author}</p>
    </section>
  );
};