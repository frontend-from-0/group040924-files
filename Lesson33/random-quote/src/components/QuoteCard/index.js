export const QuoteCard = ({ quote, author }) => {
  return (
    <section className='bg-slate-100 text-slate-800 dark:bg-slate-700 dark:text-slate-100 max-w-2xl my-5 mx-auto rounded-md p-10 text-lg'>
      <p>{quote}</p>
      <p>{author}</p>
    </section>
  );
};
