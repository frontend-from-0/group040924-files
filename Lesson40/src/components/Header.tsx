"use client";

import { useEffect, useState } from "react";
import { onAuthState, signOut } from "@/utils/firebase";
import Link from "next/link";

type UserInfo = {
  email: string;
};

export function Header() {
  const [user, setUser] = useState<UserInfo | null>(null);

  useEffect(() => {
    type Unsubscribe = () => void;

    const unsubscribe: Unsubscribe = onAuthState((user) => {
      if (user) {
        const MAX_AGE = 60 * 60 * 1000;
        document.cookie = `__session=${user.uid}; Max-Age=${MAX_AGE}; path=/`;
        setUser({ email: user.email ?? "unknown" });
      } else {
        document.cookie = `__session=; Max-Age=0; path=/`;
        setUser(null);
      }
    });

    return () => unsubscribe();
  }, []);

  return (
    <header className="w-full flex items-center justify-between px-6 py-4 border-b bg-white shadow-sm">
      <Link href="/" className="text-xl font-bold">
        Ecom
      </Link>

      <nav className="flex gap-4 items-center">
        <Link href="/cart">Cart</Link>
        {user ? (
          <>
            <span className="text-sm text-gray-600">Hello, {user.email}</span>
            <button
              onClick={signOut}
              className="bg-slate-400 text-white px-4 py-1 rounded hover:bg-slate-500"
            >
              Log Out
            </button>
          </>
        ) : (
          <Link
            href="/user/signup"
            className="bg-slate-600 text-white px-4 py-1 rounded hover:bg-slate-700"
          >
            Sign Up
          </Link>
        )}
      </nav>
    </header>
  );
}
