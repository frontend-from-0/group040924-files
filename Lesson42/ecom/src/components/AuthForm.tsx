"use client";

import { useState } from "react";

interface AuthFormProps {
  type: "signup" | "login";
  onSubmit: (email: string, password: string) => Promise<void>;
}

export function AuthForm({ type, onSubmit }: AuthFormProps) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState("");

  async function handleSubmit(e: React.FormEvent) {
    e.preventDefault();
    setError("");

    if (!email || !password) {
      setError("Email and password are required.");
      return;
    }

    if (type === "signup" && password.length < 6) {
      setError("Password must be at least 6 characters.");
      return;
    }

    try {
      await onSubmit(email, password);
    } catch (err: unknown) {
      if (err instanceof Error) {
        setError(err.message || "Something went wrong.");
      } else {
        setError("Something went wrong.");
      }
    }
  }

  return (
    <form onSubmit={handleSubmit} className="space-y-4">
      <div>
        <label htmlFor="email" className="block text-sm font-medium text-gray-700">
          Email address
        </label>
        <input
          type="email"
          id="email"
          required
          className="mt-1 block w-full rounded border-gray-300 shadow-sm focus:border-slate-500 focus:ring-slate-500"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          aria-describedby="email-description"
        />
      </div>

      <div>
        <label htmlFor="password" className="block text-sm font-medium text-gray-700">
          Password
        </label>
        <input
          type="password"
          id="password"
          required
          className="mt-1 block w-full rounded border-gray-300 shadow-sm focus:border-slate-500 focus:ring-slate-500"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        {type === "signup" && (
          <p id="password-description" className="text-xs text-gray-500 mt-1">
            Minimum 6 characters
          </p>
        )}
      </div>

      {error && <p className="text-red-600 text-sm">{error}</p>}

      <button
        type="submit"
        className="w-full bg-slate-600 text-white py-2 px-4 rounded hover:bg-slate-700"
      >
        {type === "signup" ? "Create Account" : "Log In"}
      </button>
    </form>
  );
}
