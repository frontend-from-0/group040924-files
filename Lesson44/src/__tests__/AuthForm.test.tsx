import { AuthForm, validatePassword } from '@/components/AuthForm';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const handleSubmitMock = jest.fn();

describe('AuthForm', () => {
  it('snapshot for signup form is not modified', () => {
    const { container } = render(<AuthForm type={'signup'} onSubmit={handleSubmitMock} />)
    expect(container).toMatchSnapshot();
  })

    it('snapshot for login form is not modified', () => {
    const { container } = render(<AuthForm type={'login'} onSubmit={handleSubmitMock} />)
    expect(container).toMatchSnapshot();
  })


  it('signup form renders correctly', () => {
    render(<AuthForm type={'signup'} onSubmit={handleSubmitMock} />);

    expect(screen.getByTestId('auth-form')).toBeInTheDocument();
    expect(screen.getByText('Email')).toBeVisible();
    expect(screen.getByText('Password')).toBeVisible();
    expect(screen.getByText('Create Account')).toBeVisible();
  });

  it('error message is not shown by default', () => {
    render(<AuthForm type={'signup'} onSubmit={handleSubmitMock} />);

    expect(screen.queryByTestId('form-error')).toBeNull();
  });

  it('verifies password correctly', () => {
    expect(validatePassword('asdas')).toEqual({
      success: false,
      message: 'Password must be at least 6 and max 50 characters.',
    });

    expect(validatePassword('12345    ')).toEqual({ success: true });

    const veryLongString =
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";

    expect(validatePassword(veryLongString)).toEqual({
      success: false,
      message: 'Password must be at least 6 and max 50 characters.',
    });
  });
});
