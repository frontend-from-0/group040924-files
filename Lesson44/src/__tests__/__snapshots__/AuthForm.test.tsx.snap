// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm snapshot for login form is not modified 1`] = `
<div>
  <form
    class="space-y-4"
    data-testid="auth-form"
  >
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="email"
      >
        Email
      </label>
      <input
        aria-describedby="email-description"
        class="mt-1 block w-full rounded border-gray-300 shadow-sm focus:border-slate-500 focus:ring-slate-500"
        id="email"
        required=""
        type="email"
        value=""
      />
    </div>
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="password"
      >
        Password
      </label>
      <input
        class="mt-1 block w-full rounded border-gray-300 shadow-sm focus:border-slate-500 focus:ring-slate-500"
        id="password"
        required=""
        type="password"
        value=""
      />
    </div>
    <button
      class="w-full bg-slate-600 text-white py-2 px-4 rounded hover:bg-slate-700"
      type="submit"
    >
      Log In
    </button>
  </form>
</div>
`;

exports[`AuthForm snapshot for signup form is not modified 1`] = `
<div>
  <form
    class="space-y-4"
    data-testid="auth-form"
  >
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="email"
      >
        Email
      </label>
      <input
        aria-describedby="email-description"
        class="mt-1 block w-full rounded border-gray-300 shadow-sm focus:border-slate-500 focus:ring-slate-500"
        id="email"
        required=""
        type="email"
        value=""
      />
    </div>
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="password"
      >
        Password
      </label>
      <input
        class="mt-1 block w-full rounded border-gray-300 shadow-sm focus:border-slate-500 focus:ring-slate-500"
        id="password"
        required=""
        type="password"
        value=""
      />
      <p
        class="text-xs text-gray-500 mt-1"
        id="password-description"
      >
        Minimum 6 characters
      </p>
    </div>
    <button
      class="w-full bg-slate-600 text-white py-2 px-4 rounded hover:bg-slate-700"
      type="submit"
    >
      Create Account
    </button>
  </form>
</div>
`;
